src/neuroconv/basedatainterface.py

```diff
@@ -4,25 +4,36 @@
 from typing import Optional
 
 from pynwb import NWBFile
+from pydantic.schema import model_schema
+from pydantic.decorator import ValidatedFunction
 
 from .utils import get_base_schema, get_schema_from_method_signature
 
 
 class BaseDataInterface(ABC):
     """Abstract class defining the structure of all DataInterfaces."""
 
+    @classmethod
+    def get_source_model(cls):
+        """Infer the Pydantic model for the source_data from the method signature (annotation typing)."""
+        return ValidatedFunction(function=cls.__init__, config=None).model
+
     @classmethod
     def get_source_schema(cls):
         """Infer the JSON schema for the source_data from the method signature (annotation typing)."""
-        return get_schema_from_method_signature(cls.__init__, exclude=["source_data"])
+        return model_schema(cls.get_source_model())
```

Comment:
Maintaining the JSON view, even if not directly used any more, would still be good for future web form integration I assume